packages/shared/components/molecules/TransactionDetails.svelte

```diff
@@ -0,0 +1,69 @@
+<script lang="typescript">
+    import { Pill, ActivityTypePill, Box, AddressBox, KeyValueBox } from 'shared/components/atoms'
+    import { Text } from 'shared/components'
+    import { convertToFiat, currencies, exchangeRates, formatCurrency } from 'shared/lib/currency'
+    import { formatDate, localize } from '@core/i18n'
+    import { activeProfile } from 'shared/lib/profile'
+    import { CurrencyTypes } from 'shared/lib/typings/currency'
+    import { formatUnitBestMatch, formatUnitPrecision } from 'shared/lib/units'
+    import { FontWeightText } from 'shared/components/Text.svelte'
+
+    export let timestamp = undefined
+    export let type = undefined
+    export let value = undefined
+    export let address = undefined
+    export let account = undefined
+    export let unit = undefined
+
+    let date = undefined
+    $: {
+        try {
+            if (timestamp) {
+                date = formatDate(new Date(timestamp), {
+                    dateStyle: 'long',
+                    timeStyle: 'medium',
+                })
+            }
+        } catch {
+            date = localize('error.invalidDate')
+        }
+    }
```

Comment:
I think this can be defined as a function and called reactively upon change of timestamp.

Reply:
This already has a function formatDate not sure what else to we can abstract as the handling might be different in each case. This is it in the next PR:
![Screenshot 2022-05-04 at 12 53 03](https://user-images.githubusercontent.com/89913787/166675921-f809a2d1-f3c5-4572-b2e2-6498e32a86a5.png)

Reply:
Maybe `tryFormatDate()`? I'll leave it up to you. You can resolve this if you want!